test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -731,7 +780,7 @@ private static boolean matches(ByteBuffer value, List<? extends Expression> cond
         for (Expression e : conditions)
         {
             ByteBuffer expected = eval(e);
-            if (expected.equals(value))
+            if (expected != null && expected.equals(value))
```

Comment:
`eval` is now `Nullable` but this code path shouldn't really ever see a null... maybe if you did something like `[1] - [1]` then we could get a `null` here...  mostly added to be correct and defensive (and intellij yelled at me)